client.go

```diff
@@ -1023,7 +1024,63 @@ func (cli *Client) SetAvatarURL(ctx context.Context, url id.ContentURI) (err err
 	return nil
 }
 
+// SetProfile replaces the user's entire profile.
+//
+// If MSC4133 (https://github.com/matrix-org/matrix-spec-proposals/pull/4133) is supported, this is a single PUT call.
+// Otherwise, the provided data will be parsed and the displayname and avatar are sent in separate requests.
+func (cli *Client) SetProfile(ctx context.Context, data any) (err error) {
+	return cli.setOrUpdateProfile(ctx, data, http.MethodPut)
+}
+
+// UpdateProfile updates the provided fields in the user's entire profile.
+//
+// If MSC4133 (https://github.com/matrix-org/matrix-spec-proposals/pull/4133) is supported, this is a single PATCH call.
+// Otherwise, the provided data will be parsed and the displayname and avatar are sent in separate requests.
+func (cli *Client) UpdateProfile(ctx context.Context, data any) (err error) {
+	return cli.setOrUpdateProfile(ctx, data, http.MethodPatch)
+}
+
+func (cli *Client) setOrUpdateProfile(ctx context.Context, data any, method string) (err error) {
+	if cli.SpecVersions.Supports(FeatureExtendedProfiles) || cli.SpecVersions.Supports(BeeperFeatureArbitraryProfileMeta) {
+		urlPath := cli.BuildClientURL("v3", "profile", cli.UserID)
+		_, err = cli.MakeRequest(ctx, method, urlPath, data, nil)
+	} else if cli.SpecVersions.Supports(UnstableFeatureExtendedProfiles) {
+		urlPath := cli.BuildClientURL("unstable", "uk.tcpip.msc4133", "profile", cli.UserID)
+		_, err = cli.MakeRequest(ctx, method, urlPath, data, nil)
```

Comment:
This error doesn't look returned. Did you want to do a naked return on the last line of the function?

Reply:
Oops yeah it shouldn't be nil at the end